### ISSUE TITLE
Rejected promise not throwing an uncaught exception with "no try-catch"

### ISSUE THREAD

**Issue by NaridaL** [qunitjs/qunit#1149]
## Tell us about your runtime:

* **QUnit version**: QUnit 2.0.0; Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36
* **What environment are you running QUnit in? (e.g., browser, Node)**: Chrome 57.0.2987
* **How are you running QUnit? (e.g., script, testem, Grunt)**: HTML test runner

I have a test which returns a promise which fails. I generally have no try-catch on so that I can debug errors in my IDE, but this doesn't seem to work in conjunction with promises. Is this by design or a bug?

    QUnit.test('promise fail', function (assert) {
        return new Promise((resolve, reject) => {
            resolve()
        }).then(() => {
            throw new Error("hurk")
        })
    })

HTML Reporter shows: 
Promise rejected during promise fail: hurk@ 1 ms
Source: 	
Error: hurk
    at Promise.then (http://localhost:63342/cs/out/test/PICurve.qunit.js:18:15)

**Comment by NaridaL** [comment]
Looks like a simple fix.
[promises-notrycatch.diff.txt](https://github.com/qunitjs/qunit/files/907958/promises-notrycatch.diff.txt)

**Comment by trentmwillis** [comment]
@NaridaL since it looks like you already have a solution here, would you mind opening a PR?

**Comment by leobalter** [comment]
Without signing the CLA or opening a new PR we cannot use that solution you
mentioned.

It would be great to have a PR from you, @NaridaL. :)

**Comment by NaridaL** [comment]
While my fix is an improvement, there still seem to be some problematic cases. I ended up changing it to use assert.async.

I'll take another look later and see if there's a better solution.

**Comment by NaridaL** [comment]
Turns out converting a rejected promise to an exception is pretty hard, because exceptions are automatically converted to rejected promise. One can use a settimeout (http://stackoverflow.com/a/30741722/1980909) to get out of the promise chain, but the debugger wont catch the exception then.

All of which is unecessary, because Chrome/WebStorm will catch rejected promises on their own... as long as you're using native async/await and not a transpiled polyfill. 

Anyway the fix I posted originally works. I've created a PR.
